tuplex/codegen/include/LLVMEnvironment.h

```diff
@@ -76,6 +76,12 @@ namespace tuplex {
             std::unique_ptr<llvm::Module> _module;
             std::map<python::Type, llvm::Type *> _generatedTupleTypes;
             std::map<python::Type, llvm::Type *> _generatedListTypes;
+            std::map<python::Type, llvm::Type *> _generatedIterIteratorTypes;
+            // use llvm struct member types for map key since argType may contain iterator types, and iterators with the same yieldType may have different llvm structs
+            std::map<std::vector<llvm::Type *>, llvm::Type *> _generatedZipIteratorTypes;
+            std::map<std::vector<llvm::Type *>, llvm::Type *> _generatedEnumerateIteratorTypes;
+            // string: function name; BlockAddress*: BlockAddress* to be filled in an iterator struct
+            std::map<std::string, llvm::BlockAddress *> _generatedIteratorUpdateIndexFunctions;
```

Comment:
why do we need BlockAddress here?

Reply:
It's one of the fields of list/tuple/... iterator structs.
Normally it's the address of the block that updates and checks iterator's index, but as soon as an iterator is exhausted, this field will be set to the address of a block that always returns true. When next() gets called on an exhausted iterator, it can tell whether the iterator is exhausted without having to check if index+1 > iterableObjectLength. (it's not the only way to achieve this given the current implementation and can be replaced by an additional field "i1 iteratorExhausted" in iterator struct).

Reply:
that's a great explanation, let's add it as comment?

Reply:
Added to Doxygen comment at the function (that uses this map) prototype.